packages/material/test/renderers/MaterialOneOfRenderer.test.tsx

```diff
@@ -0,0 +1,404 @@
+import React from 'react';
+import JsonRefs from 'json-refs';
+import { Provider } from 'react-redux';
+
+import Enzyme, { mount, ReactWrapper } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import { Actions, ControlElement, getData, jsonformsReducer, JsonFormsState } from '@jsonforms/core';
+import { materialFields, MaterialOneOfRenderer, materialRenderers } from '../../src';
+import { combineReducers, createStore, Store } from 'redux';
+import { JsonForms } from '@jsonforms/react';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+const initStore = () => {
+  const s: JsonFormsState = {
+    jsonforms: {
+      renderers: materialRenderers,
+      fields: materialFields
+    }
+  };
+
+  const store: Store<JsonFormsState> = createStore(
+    combineReducers({ jsonforms: jsonformsReducer() }),
+    s
+  );
+
+  return store;
+};
+
+describe('Material oneOf renderer', () => {
+
+  let wrapper: ReactWrapper;
+
+  afterEach(() => wrapper.unmount());
+
+  it('should add an item at correct path', () => {
```

Comment:
can't we unify this with the `should switch to array based oneOf subschema, then switch back, then edit` test`? imho  the second one is a better test of the first one especially if you remove the intermediate objects which is tested in the array tests

Reply:
Yes, we could do that, the schema is different though. Whatever makes more sense to you.

Reply:
I now see the different schemas, but I just tried out using one schema for both tests and they still work. The array part which is the only one being checked is the same for both schemas.